### PR TITLE
Create TagSettingsPage

### PR DESCRIPTION
**Pull request by waterim**
### Details

Tag Settings Page

### Fixed Issues

$ https://github.com/Expensify/App/issues/37313

### Tests

1. Go to workspace settings
2. Go to tags (enable in more features)
3. click on any tag
4. try to enable/disable tag
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as tests

### QA Steps

None

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [x] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Web</summary>
image
image
image

</details>
<details>
<summary>Mobile</summary>
image
image

</details>

### CODE REVIEW

File: src/libs/Navigation/linkingConfig/config.ts [hunk 1]
```diff
@@ -298,6 +298,12 @@ const config: LinkingOptions<RootStackParamList>['config'] = {
                         [SCREENS.WORKSPACE.TAG_CREATE]: {
                             path: ROUTES.WORKSPACE_TAG_CREATE.route,
                         },
+                        [SCREENS.WORKSPACE.TAG_SETTINGS]: {
+                            path: ROUTES.WORKSPACE_TAG_SETTINGS.route,
+                            parse: {
+                                tagName: (tagName: string) => encodeURIComponent(tagName),
```
Comment (luacmartins) [inline src/libs/Navigation/linkingConfig/config.ts:304]: We need to decode the route here
```suggestion
                                tagName: (tagName: string) => decodeURIComponent(tagName),
```

**Review by luacmartins — CHANGES_REQUESTED** [review]
Looking good. Left a few comments

File: src/pages/workspace/tags/TagSettingsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    style={[styles.defaultModalContainer]}
+                    testID={TagSettingsPage.displayName}
+                >
+                    <HeaderWithBackButton title={route.params.tagName} />
+                    <View style={styles.flexGrow1}>
+                        <OfflineWithFeedback
+                            errors={ErrorUtils.getLatestErrorMessageField(currentPolicyTag)}
+                            pendingAction={currentPolicyTag.pendingFields?.enabled}
+                            errorRowStyles={styles.mh5}
+                            onClose={() => {}}
```
[the remark this answers is not in the document]
Comment (luacmartins) [reply src/pages/workspace/tags/TagSettingsPage.tsx:61]: Ah I see we're introducing that [here](https://github.com/Expensify/App/pull/38332/files#diff-cd0dfd9f8af5bc7cd33ff4f0337ef70574d42424ceecd7692b193348d3d8ecd4R2890). Maybe we can wait for it and then update this on a follow up

[the remark this answers is not in the document]
Comment (luacmartins) [reply src/pages/workspace/tags/TagSettingsPage.tsx:61]: Yes, let's add it here

File: src/libs/actions/Policy.ts [hunk 3]
```diff
@@ -2887,6 +2889,92 @@ function createPolicyTag(policyID: string, tagName: string) {
...
+                                    ...policyTag.tags[key],
+                                    ...tagsToUpdate[key],
+                                    errors: null,
+                                    pendingFields: {
+                                        enabled: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                                    },
+                                };
+
+                                return acc;
+                            }, {}),
```
[the remark this answers is not in the document]
Comment (waterim) [reply src/libs/actions/Policy.ts:2914]: Thats was added for a future use when you can select from a selectionList multiple rows and make it “enable” and “disable”

[the remark this answers is not in the document]
Comment (luacmartins) [reply src/libs/actions/Policy.ts:2914]: Yea, we gotta handle the bulk action too, so this makes sense

**Comment by luacmartins** [comment]
@waterim let me know when you address the comments and resolve conflicts. Let's get this merged!

File: src/libs/actions/Policy.ts [hunk 4]
```diff
@@ -2887,6 +2889,92 @@ function createPolicyTag(policyID: string, tagName: string) {
...
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    [policyTag.name]: {
+                        tags: {
+                            ...Object.keys(tagsToUpdate).reduce<PolicyTags>((acc, key) => {
+                                acc[key] = {
+                                    ...policyTag.tags[key],
+                                    ...tagsToUpdate[key],
```
[the remark this answers is not in the document]
Comment (waterim) [reply src/libs/actions/Policy.ts:2943]: Because of the reduce I cant remove it because of types. Same we have with categories

**Comment by waterim** [comment]
@luacmartins Resolved conflict and updated

**Comment by waterim** [comment]
@hoangzinh damn,checking

**Comment by waterim** [comment]
Weird, it works for me

**Comment by waterim** [comment]
Hmm, okay, it doesnt work for newly created tags

**Comment by waterim** [comment]
Fixed, pushed

**Comment by waterim** [comment]
It was because of the removal from reduce data regarding the tags to update, reduce is generating new object, thats why we need to pass it as well

File: src/libs/actions/Policy.ts [hunk 5]
```diff
@@ -2900,6 +2902,91 @@ function createPolicyTag(policyID: string, tagName: string) {
...
+                value: {
+                    [policyTag.name]: {
+                        tags: {
+                            ...Object.keys(tagsToUpdate).reduce<PolicyTags>((acc, key) => {
+                                acc[key] = {
+                                    ...policyTag.tags[key],
+                                    errors: null,
+                                    pendingFields: {
+                                        enabled: null,
+                                    },
```
[the remark this answers is not in the document]
Comment (waterim) [reply src/libs/actions/Policy.ts]: we can leave as now, as categories done, to have consistency

**Review by hoangzinh — APPROVED** [review]
LGTM

**Review by luacmartins — APPROVED** [review]

File: src/pages/workspace/tags/TagSettingsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    /** All policy tags */
+    policyTags: OnyxEntry<PolicyTagList>;
+};
+
+type TagSettingsPageProps = TagSettingsPageOnyxProps & StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.TAG_SETTINGS>;
+
+function TagSettingsPage({route, policyTags}: TagSettingsPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const policyTag = useMemo(() => PolicyUtils.getTagList(policyTags, 0), [policyTags]);
```
Comment (DylanDylann) [inline src/pages/workspace/tags/TagSettingsPage.tsx:36]: Crash App from this line 


https://github.com/Expensify/App/assets/141406735/2afba092-010d-4808-948f-bb4e27911a5e

**Pull request merged**